### PR TITLE
Github Actions PoC

### PR DESCRIPTION
**Pull request by Zejnilovic**
Made a small PoC for GitHub actions
This only runs the tests, integration tests and scala-style
It runs it in parallel
Caches maven dependencies

### CODE REVIEW

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
```
Comment (benedeki) [inline .github/workflows/unit_test.yml:20]: What dos this line doing?

File: pom.xml [hunk 2]
```diff
@@ -365,6 +366,7 @@
                         </goals>
                         <configuration>
                             <suffixes>(?&lt;!IntegrationSuite)</suffixes>
+                            <skipTests>${skip.unit.tests}</skipTests>
```
Comment (benedeki) [inline pom.xml:369]: To understand, why was this one needed?

File: .github/workflows/unit_test.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+name: Run-Tests
```
Comment (benedeki) [inline .github/workflows/unit_test.yml:1]: The file name is `unit_test.yaml`. But this does three types of tests.
Maybe let's rename this to something better describing this complexity and `unit_test.yaml` have the `unit-test` job?

File: .github/workflows/unit_test.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-
+      - name: Run the scalastyle maven plugin
+        run: mvn -B scalastyle:check
+  unit-test:
```
Comment (benedeki) [inline .github/workflows/unit_test.yml:25]: Tiny: Shouldn't be `unit-tests` better to show, it's a battery of tests?

File: .github/workflows/unit_test.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-
+      - name: Run the Unit tests
+        run: mvn -B test
+  integration-test:
```
Comment (benedeki) [inline .github/workflows/unit_test.yml:42]: Ditto on plural.

File: .github/workflows/unit_test.yml (hunk 1 above)
Comment (Zejnilovic) [reply .github/workflows/unit_test.yml:20]: caches the dependencies, this is the key it caches them by.

https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows#input-parameters-for-the-cache-action

File: pom.xml (hunk 2 above)
Comment (Zejnilovic) [reply pom.xml:369]: So I can run parallel integration tests and unit tests

File: .github/workflows/unit_test.yml (hunk 3 above)
Comment (Zejnilovic) [reply .github/workflows/unit_test.yml:1]: the file name is bad. It was when my first PoC had 4 files. Ended with one and forgot to rename it. I would rename the file to `pr_tests.yml`. How about that?

File: pom.xml (hunk 2 above)
Comment (benedeki) [reply pom.xml:369]: Sorry for being slow, how does this enable that? Can you override the pom property in the yaml file of the action?

File: .github/workflows/unit_test.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-
+      - name: Run the scalastyle maven plugin
+        run: mvn -B scalastyle:check
```
Comment (benedeki) [inline .github/workflows/unit_test.yml:24]: We have a ton of warnings there. Won't they cause issues?

File: pom.xml (hunk 2 above)
Comment (benedeki) [reply pom.xml:369]: Or is `true` the default?

Comment (Zejnilovic) [reply pom.xml:369]: false is the default. False needs to be set using `-Dskip.unit.tests=true`. That is used in integration tests.

File: .github/workflows/unit_test.yml (hunk 6 above)
Comment (Zejnilovic) [reply .github/workflows/unit_test.yml:24]: Warnings are ok. I think. It is failing on dependencies

File: pom.xml (hunk 2 above)
Comment (benedeki) [reply pom.xml:369]: Understood now. The property is overridden in the job in the file above.

**Review by dk1844 — APPROVED** [review]

**Comment by Zejnilovic** [comment]
Doesn't make sense to do until we solve the integration tests and even the external dependencies.

**Pull request closed without merging**